core/transaction.go

```diff
@@ -165,42 +194,38 @@ type DeclareTransaction struct {
 	Version *felt.Felt
 }
 
-func (d *DeclareTransaction) hash() *felt.Felt {
-	return d.Hash
+func (d *DeclareTransaction) Hash() *felt.Felt {
+	return d.TransactionHash
 }
 
-func (d *DeclareTransaction) signature() []*felt.Felt {
-	return d.Signature
+func (d *DeclareTransaction) Signature() []*felt.Felt {
```

Comment:
To align it with it's hash counterpart made public.

Reply:
Its hash counterpart was made public so that we can reference it in `blockchain`. Imo, `Signature` is not needed outside the `transaction` module. So, it's not necessary to do that. 

I think it's fine to have both private and public functions in the same interface

Reply:
RPC will use Signature() soon. It was naive to assume that we wont use the interface outside of `core` in the first place anyways.